pkg/controller/machineset/delete_policy.go

```diff
@@ -67,6 +76,34 @@ func oldestDeletePriority(machine *v1alpha1.Machine) deletePriority {
 	return deletePriority(float64(mustDelete) * (1.0 - math.Exp(-d.Seconds()/secondsPerTenDays)))
 }
 
+
+func customDeletePolicy(machine *v1alpha1.Machine) deletePriority {
+	// Iterate through conditions to get if machine is not "Ready"
+	// Check types.go
+	for _, c := range machine.Status.Conditions {
+		if c.Type != corev1.NodeReady {
+			return mustDelete
```

Comment:
This check doesn't work.
It would trigger for any other condition that's not `NodeReady`, so potentially just all machines equally.
We should explicitly find the `NodeReady` condition and check if it's false. Then prefer it.

Reply:
Sure, I'm doing some changes to this, because it doesn't make sense as it is. I'm checking if it's possible to extract from here (extracted from .status.conditions):

```json
{
    "lastHeartbeatTime": "2024-03-26T09:15:42Z",
    "lastTransitionTime": "2024-03-25T10:06:24Z",
    "message": "kubelet is posting ready status. AppArmor enabled",
    "reason": "KubeletReady",
    "status": "True",
    "type": "Ready"
  }
```

The "status" condition (expecting True) to be more accurate with the node status too.